sorc/chgres_cube.fd/input_data.F90

```diff
@@ -6681,4 +6718,226 @@ subroutine check_cnwat(cnwat)
   enddo
 end subroutine check_cnwat
 
+
+
+
+!> Pressure to presure vertical interpolation for tracers with linear or lnP
+!> interpolation. Input tracers on pres levels are interpolated 
+!> to the target output pressure levels. The matching levels of input and 
+!> output will keep the same. Extrapolation is also allowed but needs
+!> caution. The routine is mostly for GFSV16 combined grib2 input when spfh has
+!> missing levels in low and mid troposphere from U/T/HGT/DZDT. 
+!!
+!! @param ppin     [in] 1d input pres levs
+!! @param xxin     [in] 1d input tracer
+!! @param npin     [in] number of input levs 
+!! @param ppout    [in] 1d target pres levs 
+!! @param xxout    [out] 1d interpolated tracer
+!! @param npout    [in] number of target levs 
+!! @param linlog   [in] interp method.1:linear;not 1:log;neg:extrp allowed
+!! @param xmsg     [in] fill values of missing levels (-999.0)
+!! @param ier      [out] error status. non 0: failed interpolation
+!! @author NCL code for pressure level interpolation
+!! @author adopted by Jili Dong EMC for chgres_cube 
+
+SUBROUTINE DINT2P(PPIN,XXIN,NPIN,PPOUT,XXOUT,NPOUT   &
+                      ,LINLOG,XMSG,IER)
+      IMPLICIT NONE
+!
+! This code was designed for one simple task. It has since
+! been mangled and abused for assorted reasons. For example,
+! early gfortran compilers had some issues with automatic arrays.
+! Hence, the C-Wrapper was used to create 'work' arrays which
+! were then passed to this code.  The original focused (non-NCL) 
+! task was to handle PPIN & PPOUT that had the same 'monotonicity.' 
+! Extra code was added to handle the more general case. 
```

Comment:
This statement does not give me much confidence.

Reply:
> This statement does not give me much confidence.

Same feeling when I first read this. After looking through the code, I didn't see anything suspicious. Given that this NCL code has been used for a while and is quite simple, it should be ok for our applications here. Hope unit test can also confirm this too.

Reply:
I don't think the failure is anything you did. There is a rerun-jobs button you can click on.

Reply:
> I don't think the failure is anything you did. There is a rerun-jobs button you can click on.

I can't find the re-run button. Or I don't have access/permission to it?

Reply:
Click on the failed test "details". You will see the button at the upper right.

Reply:
> Click on the failed test "details". You will see the button at the upper right.

still can't find it. Please see attached screenshots.

https://drive.google.com/file/d/1wfQbj77Ug4YyJL91Tj3zZrUBisDpC05c/view?usp=sharing
https://drive.google.com/file/d/1U4VcTEiruhTNCi0K-dFYOkikwpeMXtiL/view?usp=sharing

Reply:
Looks like you don't have permissions. I will rerun for you.

Reply:
> Looks like you don't have permissions. I will rerun for you.

Thank you @GeorgeGayno-NOAA Now it all passed.

Reply:
Great. Next I will add a sample unit test to your branch.